lib/pool.js

```diff
@@ -355,17 +355,19 @@ class WorkerNodes extends EventEmitter {
 
     /**
      * Take Heap Snapshot and save result on main process directory
-     *
-     * @returns {void}
+     * @param {TakeSnapshotOptions} options
+     * @returns {Promise<string>}
      */
-    takeSnapshot() {
+    takeSnapshot(options = {}) {
         const worker = this.pickWorker();
 
         if (worker) {
-            worker.takeSnapshot();
+            return worker.takeSnapshot(options);
         } else {
-            // There might not be availble worker, let it start.
-            setTimeout(() => this.takeSnapshot(), 500);
+            return new Promise((resolve) => {
+                // There might not be available worker, let it start.
+                setTimeout(() => resolve(this.takeSnapshot()), 500);
```

Comment:
shouldn't we pass the same options to the second call?

```suggestion
                setTimeout(() => resolve(this.takeSnapshot(options)), 500);
```